terraso_backend/apps/auth/middleware.py

```diff
@@ -23,56 +29,82 @@
 
 
 class JWTAuthenticationMiddleware:
-    def __init__(self, get_response):
-        self.get_response = get_response
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        auth_optional = getattr(view_func, "auth_optional", False)
+        auth_required = not self._is_path_public(request.path) and not auth_optional
```

Comment:
question: does Python do short-circuit conditionals? If so, put `not auth_optional` first, since it'll be quicker.